### ISSUE TITLE
Idea: Decouple from Sentry logging service

### ISSUE THREAD

**Issue by bedeho** [Joystream/atlas#3241]
# Background
Currently, logging of errors relies on the [Sentry](https://sentry.io) logging service, which is a closer logging infrastructure. Dependency happens specifically through use of their client side loggin library or interacting with their infrastructure, but its very well isolated in the overall codebase. Some operators may not want or be able to use this service, in particular when it starts to scale, hence it would be good if there was more flexibility here.

# Proposal
Abstract the logging transport in a way where one can use different logging backends, including of course Sentry. Could be done at a variety of ways at the code level, so implementer should outline some alternatives. One valuable property would be if there could be set of pluggable integrations that people could use which allowed them to reuse the work of others to make he integration work with some service. Sentry would just be one of those.

# Scope
* [ ] With this issue we will add some generic interface for loggers and ability to configure via some file and then community will be free to contribute integrations for specific services

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201958687417145/1203088857184971) by [Unito](https://www.unito.io)

**Comment by kdembler** [comment]
@bedeho I'm not sure what is the goal of this issue. The app isn't really couple with Sentry - we have a logging abstraction and all Sentry interaction happens in this specific file - this way if somebody wants to use something else it's enough to update the logger implementation, without touching other parts of the codebase. What further improvement would you like to see?

**Comment by bedeho** [comment]
Its always my mantra to try to make the app somewhat customizable without having to start to code _to get started_. So the idea in this case would be that over time, perhaps people write integrations for different backends, like Mixpanel, google analytics etc. It would be great if the next person to come around can just pick one of the existing integrations off the shelf without having to do any coding. The way it sounds right now, it would perpetually be the case that everyone would have to fork and make changes only to make one of these services work.

I don't think this is a big problem, but seems like it would be useful longer term. In some sense, it may actually not be that big of a deal precisely because exactly when people are just starting out, they may not really care all that much about whether they use this or that backend (e.g. because there is some free tier).

**Comment by traumschule** [comment]
Searching for `sentry` in this repo brings up many files importing `SentryLogger`. Is the goal of this issue to replace this with a more abstract logger that can be configured via vite or `config.yml` to use sentry or for example an ES instance hosted by jsstats.

**Comment by kdembler** [comment]
@traumschule Yeah, we would rename that to something generic (like `OnlineLogger` or whatever). 

> Is the goal of this issue to replace this with a more abstract logger that can be configured via vite or config.yml to use sentry or for example an ES instance hosted by jsstats.

I would be happy to offer configuration like this, but someone would actually need to write the specific logger implementations. Let's say with this issue we will add some generic interface for loggers and ability to configure via some file and then community will be free to contribute integrations for specific services?

**Comment by bedeho** [comment]
> Let's say with this issue we will add some generic interface for loggers and ability to configure via some file and then community will be free to contribute integrations for specific services?

Sounds good to me, I think conclusion is more important than doing it early here. As long as Sentry has free accounts we don't need to prior this, we just need have it in mind as one of the tweaks to make Atlas more configurable.

**Comment by traumschule** [comment]
bumping for #4008 and #3997

The goal is to let operators decide where and how to log. The logger must be agnostic of the backend and just handle "spooling" of messages to any enabled _logger implementations (plugins).
Use case: As a good default the app could log (resolution failures) to a configurable WG-run message cache.

**Comment by bedeho** [comment]
I don't believe sentry logging has anything to do with those issues (#4008 and #3997), Sentry is for logging application specific private issues, while those other issues are for logging into public DAO level infra structure.

**Comment by bedeho** [comment]
See sentry box in this shot from https://gleev.xyz/video/1302?time=828

![Screenshot 2023-04-03 at 10 53 58](https://user-images.githubusercontent.com/437292/229460852-0f526cc9-90c1-4135-94aa-5809bdec4de9.png)
